frontend/src/routes/Governance/components/ViewDiffApiCall.tsx

```diff
@@ -21,26 +22,47 @@ export function ViewDiffApiCall({ item }: Readonly<{ item: ResultsTableData }>)
     setIsModalOpen((prevIsModalOpen) => !prevIsModalOpen)
   }
   useEffect(() => {
+    let ignore = false
+
     if (isModalOpen) {
       setIsFetching(true)
       setRelatedObjs(undefined)
       setTemplateErr(undefined)
 
-      fireManagedClusterView(cluster, kind, apiVersion, templateName, cluster)
+      const viewName = process.env.NODE_ENV === 'test' ? undefined : uuidv4()
+
+      fireManagedClusterView(cluster, kind, apiVersion, templateName, cluster, viewName, viewName !== undefined)
```

Comment:
Our goal was to reduce the frequency of hitting fireManagedClusterView. But this still hitting fireManagedclusteview when ignore is true. ?  should more if ignore return to before firemanaged?

Reply:
The goal to reduce the `fireManagedClusterView` was in reducing the rerenders. The `ignore` variable is so that if the effect is reexecuted, the old promise results are ignored and don't affect the UI results.